src/app/server/Mdns.cpp

```diff
@@ -122,28 +122,23 @@ CHIP_ERROR AdvertiseOperational()
 /// Set MDNS commissioner advertisement
 CHIP_ERROR AdvertiseCommisioner()
 {
-    return Advertise(false);
+    return Advertise(false, false, false);
 }
 
 /// Set MDNS commissionable node advertisement
-CHIP_ERROR AdvertiseCommissionableNode()
+CHIP_ERROR AdvertiseCommissionableNode(bool commissioningMode, bool additionalCommissioning)
 {
-    return Advertise(true);
+    return Advertise(true, commissioningMode, additionalCommissioning);
 }
 
-/// commissionableNode
-// CHIP_ERROR Advertise(chip::Mdns::CommssionAdvertiseMode mode)
-CHIP_ERROR Advertise(bool commissionableNode)
+/// Set MDNS commissioner advertisement or commissionable node advertisement
+CHIP_ERROR Advertise(bool commissionableNode, bool commissioningMode, bool additionalCommissioning)
```

Comment:
The callsites of this are pretty unreadable... Should this take a bitfield arg instead, so callsites would list the bits that are true and it would be clearer what is being advertised?

Reply:
added bitmap